src/compile/data/optimizers.ts

```diff
@@ -78,3 +78,29 @@ export function removeUnusedSubtrees(node: DataFlowNode) {
   }
   return true;
 }
+
+/**
+ * Generates time unit nodes from selection filters and removes duplicate
+ * time unit nodes (as determined by the name of the output field)
+ * that may be generated due to selections projected over time units.
+ */
+export function cleanupSelectionTimeUnits(leaf: DataFlowNode) {
+  let fields = {};
+  return iterateFromLeaves((node: DataFlowNode) => {
+    if (node instanceof FilterNode) {
+      // Selection filters only add TimeUnitNodes on assembly.
```

Comment:
Why is this the case? Can we not generate the nodes when we parse or assemble the selection? 

Can you walk me through the parse and assemble steps for selections? When do we know what timeUnit a selection uses? Do we have access to the selection filter dataflow node in the assembly phase of the selection?